packages/utils/src/common/create-psql-index-helper.ts

```diff
@@ -35,17 +37,38 @@ export function createPsqlIndexStatementHelper({
   where,
   unique,
 }: {
-  name: string
+  name?: string
   tableName: string
   columns: string | string[]
   type?: string
   where?: string
   unique?: boolean
 }) {
+  const columnsName = Array.isArray(columns) ? columns.join("_") : columns
+
   columns = Array.isArray(columns) ? columns.join(", ") : columns
+  name = name || `IDX_${tableName}_${columnsName}`
+
   const typeStr = type ? ` USING ${type}` : ""
   const optionsStr = where ? ` WHERE ${where}` : ""
   const uniqueStr = unique ? "UNIQUE " : ""
 
-  return `CREATE ${uniqueStr}INDEX IF NOT EXISTS "${name}" ON "${tableName}"${typeStr} (${columns})${optionsStr}`
+  const expression = `CREATE ${uniqueStr}INDEX IF NOT EXISTS "${name}" ON "${tableName}"${typeStr} (${columns})${optionsStr}`
+  return {
+    toString: () => {
+      return expression
+    },
+    valueOf: () => {
+      return expression
+    },
+    name,
+    expression,
+    MikroORMIndex: (options = {}) => {
+      return Index({
+        name,
+        expression,
+        ...options,
+      })
+    },
+  }
```

Comment:
The idea is to be able to construct the index object and immediately apply it to the props without having to explicitly declared a name and build the expression and then having to apply them all.

```ts
const Index = createPsqlIndexHelper({
  name: "IDX",
  ...
})

class Entity {
  @Index.MikroORMIndex()
  prop: something
}
```

Reply:
Sorry, I realise I phrased my question wrong. Why do we need the different return values and why are we not using it the same why in the modules?

Reply:
You can have a need for the values for example like in the customer module where you need to index name in order to handle the error manually, in that case this index is exported and you can extract the name from it. The expression could be also needed why not. The idea was to not fully changed the other modules in order to reduce the amount of conflict. So the other modules are just extracting the expression so that only the indexes are touched and not the entities

Reply:
I didn't want to change the default behavior of the helper, and regardless the ORM used, the expression in plain sql might be needed.
I will refactor all the other modules in a next PR. as Adrien said, I'm avoiding a big number of conflicts for now.